### ISSUE TITLE
delocate.delocating.DelocationError: .dylibs/libgfortran.5.dylib already exists

### ISSUE THREAD

**Issue by letmaik** [matthew-brett/delocate#53]
This worked in the past but suddenly I get an error when running `delocate-path` on a folder where multiple executables rely on the same dependency (I'm assuming that's what the error is about): "delocate.delocating.DelocationError: .dylibs/libgfortran.5.dylib already exists". Am I doing something wrong?

**Comment by letmaik** [comment]
@matthew-brett What can I do to help you analyze the problem?

**Comment by matthew-brett** [comment]
Yes, sorry - can you make me some way of replicating the problem?

**Comment by letmaik** [comment]
@matthew-brett I couldn't reproduce it locally yet, only on CI weirdly. Will investigate further. But if there's any verbose logging etc I can enable, let me know.

**Comment by matthew-brett** [comment]
Can you point me to the error on CI log?

**Comment by letmaik** [comment]
https://dev.azure.com/wrf-cmake/WRF/_build/results?buildId=428 See the failed macOS build and click on the "Delocate" step.

**Comment by letmaik** [comment]
I [dumped out](https://dev.azure.com/WRF-CMake/WRF/_build/results?buildId=482&view=logs&jobId=571f6113-b177-504f-680c-6f7e5f53d965&taskId=d2662577-9dbc-5143-648b-0cd8f182b67e&lineStart=30&lineEnd=110&colStart=1&colEnd=31) `delocate-listdeps` in case it's helpful.

**Comment by letmaik** [comment]
OK, I found the issue.

`libgfortran.5.dylib` was a direct dependency of the executables listed above, but also an indirect dependency via `libnetcdff.6.1.1.dylib` (which is *not* listed in `delocate-listdeps`). The catch was that the direct dependency came from `/usr/local/Cellar/gcc@8/8.3.0/lib/gcc/8/libgfortran.5.dylib` and the indirect one from `/usr/local/Cellar/gcc@9/9.1.0/lib/gcc/9/libgfortran.5.dylib`. delocate then couldn't handle the situation of different absolute paths with the same filename. I changed my build to use the same gfortran for everything which solved the issue.

There are two things that should be improved I think:
- Show indirect dependencies in `delocate-listdeps`.
- Improve the error message for this situation and include the absolute paths of the two conflicting dylibs.

**Comment by HexDecimal** [comment]
PR #94 should have improved situations with indirect dependencies.  Can you test your wheel with the latest version of delocate?

`delocate-listdeps` also shows indirect dependencies since #118 was merged.
